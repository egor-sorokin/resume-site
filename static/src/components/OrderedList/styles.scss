@import '../../styles/base/_vars.scss';

.orderedlist {
  &__list {
    list-style: none;
    counter-reset: item - 1;
  }

  &__item {
    counter-increment: item;
    position: relative;
    display: flex;
    flex-direction: row;


    &:before {
      content: counter(item, decimal-leading-zero);
      margin-left: 10px;

      min-width: 20px;
    }
    &:after {
      content: '';
      display: block;

      position: absolute;
      top: 14px;
      left: 0;

      width: 5px;
      height: 1px;

      background-color: $mercuryLight;
    }
    &-dash {
      margin-top: 13px;
      margin-right: 10px;

      width: 15px;
      height: 1px;

      background-color: $mercuryLight;
    }
  }
  &__text {
    color: $mercuryLight;
  }
  &__title {
    font-family: $secondaryFont;
    font-size: 12px;
    letter-spacing: 3px;
    text-align: left;
    text-transform: uppercase;
    font-weight: 700;
  }
}
