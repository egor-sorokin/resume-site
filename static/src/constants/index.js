export const DEFAULT_API_URL = 'http://127.0.0.1:8080/api/v.1.0/';
export const URL_PATH_PERSON_DATA = 'person';
export const URL_PATH_PROJECTS = 'projects';
