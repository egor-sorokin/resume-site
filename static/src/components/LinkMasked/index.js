import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = ({linkData, cssClasses, target}) => (
  <div className="link-masked">
    <span className={`link-masked__placeholder ${cssClasses}`}>{linkData.text}</span>
    <div className="link-masked__mask">
      <a
        href={linkData.url}
        target={target}
        className={`link-masked__text ${cssClasses}`}
      >
        {linkData.text}
      </a>
    </div>
  </div>
);

LinkStretched.propTypes = {
  linkData: PropTypes.shape({}),
  cssClasses: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  linkData: {},
  cssClasses: '',
  target: ''
};

export default LinkStretched;