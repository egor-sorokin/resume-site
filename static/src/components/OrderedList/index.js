import React from 'react';
import './styles.scss';


const OrderedList = () => {
  const orderedListItems = ['LinkedIn', 'Xing', 'GitHub', 'Codepen'].map((item) => (
    <li className="orderedlist__item">
      <span className="orderedlist__text">{item}</span>
    </li>
  ));

  return (
    <div className="orderedlist">
      <h5 className="orderedlist__title">Attachments</h5>
      <ol className="orderedlist__list" start="1">
        {orderedListItems}
      </ol>
    </div>
  )
};


export default OrderedList;