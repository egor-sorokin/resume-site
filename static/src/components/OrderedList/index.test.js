// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import OrderedList from './index';
import LinkMasked from '../LinkMasked/index';
import {LINK_MASKED} from '../../constants/index';


describe('<OrderedList />', () => {
  let component;
  const title = 'title';
  const items = [
    {
      id: 1,
      text: 'test text 1',
      url: 'test.urlone'
    },
    {
      id: 2,
      text: 'test text 2',
      url: 'test.urltwo'
    }
  ];


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<OrderedList />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 component', () => {
      expect(component.children()).toHaveLength(1);
    });
  });


  describe('props tests', () => {
    beforeEach(() => {
      component = mount(<OrderedList items={items} />);
    });

    it('renders 1 simple link', () => {
      expect('items' in component.props()).toBeTruthy();
      expect(component.find('li').first().childAt(1).is('a')).toBeTruthy();
    });


    it('renders masked link', () => {
      component.setProps({type: LINK_MASKED});
      component.update();

      expect('items' in component.props()).toBeTruthy();
      expect('type' in component.props()).toBeTruthy();
      expect(component.prop('type')).toEqual(LINK_MASKED);
      expect(component.find(LinkMasked)).toHaveLength(2);
    });


    it('renders a title for list', () => {
      component.setProps({title: title});
      expect('items' in component.props()).toBeTruthy();
      expect('title' in component.props()).toBeTruthy();
      expect(component.prop('title')).toEqual(title);
      expect(component.find('.orderedlist__title')).toHaveLength(1);
    });
  });
});
