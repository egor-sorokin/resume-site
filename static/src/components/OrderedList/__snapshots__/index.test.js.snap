// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderedList /> base tests matches snapshot 1`] = `
<div
  className="orderedlist"
>
  <ol
    className="orderedlist__list"
  />
</div>
`;
